### DISCUSSION TITLE
Resulting plot by torchio.Subject.plot() is looking strange. Why?

### DISCUSSION THREAD

**Discussion by cepa995** [TorchIO-project/torchio#652] [Q&A] [answered]
Hi,

I am really amazed with torchio as a toolkit for medical image analysis, but I am having trouble understanding what am I doing wrong when trying to plot subject images.

For example, I have a simple function:

```python
    def plot_subject(self, subject_id=None):
        """ Plots subject that corresponds to the specified subject ID, or
        random if not specfiied..
        
        :param subject_id - string by which subjects are uniquely identified.
        """
        if subject_id:
            subject_idx = [idx for idx, subject in enumerate(self.data.dry_iter(), 0)
                                       if subject.__dict__['subject_id'] == subject_id][0]
            if not subject_idx:
                raise ValueError(f"[ERROR]: Subject with {subject_id} does not exist in the dataset!")
        else:
            subject_ids = [subject.__dict__['subject_id'] for subject in self.data.dry_iter()]
            subject_idx = random.randint(0, len(subject_ids))
        subject = self.data[subject_idx]
        images = subject.get_images()
        for img in images:
            print(f"[INFO]: {img.shape}")
        print(f"[INFO]: Plotting subject with ID - {self.data[subject_idx].__dict__['subject_id']}")
        self.data[subject_idx].plot()This takes REALLY long time to process. 

```
**NOTE**: self.data is` torchio.SubjectsDataset`, also I am **Resampling** all images (there are 4x images) to "T2w", which is name of one of the images!

This is what I get as the output:

![image](https://user-images.githubusercontent.com/67524891/132121460-956111bb-ffaa-4622-9fca-e3ee6fc8a141.png)

What is strange is that I am getting different results for different patients:

![image](https://user-images.githubusercontent.com/67524891/132121516-75a3fdf3-ab49-455f-b62b-87ea4416e414.png)

![image](https://user-images.githubusercontent.com/67524891/132121722-502f1d5d-11d8-4996-8477-878c2fd50761.png)

My question is, am I doing something wrong here, is this maybe data related, or? Here is list of transforms being applied:

```
tio.Compose([
                tio.ToCanonical(),
                tio.Resample("T2w"),
                tio.RescaleIntensity((-1, 1)),
                tio.OneHot(),
            ])
```

**Comment by fepegar** [comment, marked as answer]
I've added a new dataset class and written a notebook motivated by your questions, it might be useful for you: https://www.kaggle.com/fepegar/preprocessing-mri-with-torchio

I now have the dataset, so I can try plotting a troublesome subject myself and diagnose the issue.

**Reply by fepegar** [reply]
Also, I recently changed the plotting method and I think it's still behaving a bit strangely for subjects with more than two images.

**Reply by fepegar** [reply]
I can reproduce the ugly plotting with the following code (only subject 00792 is present in the dataset folder):

```python
import torchio as tio
dataset = tio.datasets.RSNAMICCAI('root')
dataset[0].plot()
subject = dataset[0]
subject.shape
dataset[0].plot()
transform = tio.Compose([
                tio.ToCanonical(),
                tio.Resample("T2w"),
                tio.RescaleIntensity((-1, 1)),
                tio.OneHot(),
            ])
transform(subject).plot()
```

**Reply by cepa995** [reply]
Ah, that's great. Thanks @fepegar .

Strange thing is that this is not true for all participants. This leads me to believe it might be some data related issue. I could be wrong though. If you have any luck figuring this out please let me know. I will do the same.

**Reply by fepegar** [reply]
I think this is a visualization issue and it will hopefully be solved after merging #653.

**Reply by cepa995** [reply]
Hi @fepegar ! This fixed worked like a charm. I appreciate for investing time into it, and fixing it asap.

**Reply by fepegar** [reply]
No worries! Thanks for reporting.
